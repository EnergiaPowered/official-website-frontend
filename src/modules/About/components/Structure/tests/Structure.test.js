import React from "react";
import ReactDOM from "react-dom";

import Structure from "../index";

import { create } from "react-test-renderer";
import { act } from "react-dom/test-utils";

import { cleanup, render, fireEvent } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

// matches snapshot
it("matches snapshot", () => {
  const component = create(<Structure />);

  expect(component.toJSON()).toMatchSnapshot();
});

// image zoom on click
it("state changes on click", () => {
  const { getByTestId } = render(<Structure />);

  expect(
    getByTestId("structure-image").dataset.teststate === "true"
  ).toBeFalsy();

  fireEvent.click(getByTestId("structure-image"));

  expect(
    getByTestId("structure-image").dataset.teststate === "true"
  ).toBeTruthy();
});
