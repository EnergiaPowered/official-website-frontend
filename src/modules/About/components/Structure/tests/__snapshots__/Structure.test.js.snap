// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="bg-section about-structure"
>
  <h2
    className="section-title"
  >
    Our Structure
  </h2>
  <div
    className="d-flex justify-content-center"
  >
    <div
      className="EnlargedImage"
    >
      <div
        className="EnlargedImage__container"
        style={
          Object {
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <img
          alt="Our team's structure"
          className="EnlargedImage__Image"
          data-testid="structure-image"
          data-teststate={false}
          onClick={[Function]}
          src="Structure.jpg"
          style={
            Object {
              "cursor": "zoom-in",
              "height": "auto",
              "maxWidth": "100%",
              "opacity": 1,
            }
          }
          title="Energia Powered'20 board"
        />
      </div>
    </div>
  </div>
</section>
`;
